app/models/event_context.rb

```diff
@@ -1,8 +1,16 @@
 class EventContext < ActiveSupport::CurrentAttributes
   attribute :request_event
+  attribute :events_suppressed
+
+  # Stop sending events in the current context (useful for e.g. background jobs where records are
+  # updated programatically)
+  def suppress_events!
+    self.events_suppressed = true
```

Comment:
How does it know only to suppress events in the background job?

Reply:
The background job (or anywhere else that you would want to not send events) explicitly calls `suppress_events!`.

But I've found an issue with it that I'm about to fix that will also make this clearer (hopefully) 🤞